sdk/core/azure-core/src/main/java/com/azure/core/http/ProxyOptions.java

```diff
@@ -468,4 +455,71 @@ private static String sanitizeNonProxyHosts(String[] nonProxyHosts) {
             return proxyType;
         }
     }
+
+    private static class Properties {
+        public static final ConfigurationProperty<String> NO_PROXY = ConfigurationProperty.stringPropertyBuilder("http.proxy.non-proxy-hosts")
+            .environmentAliases("http.nonProxyHosts", Configuration.PROPERTY_NO_PROXY)
+            .canLogValue(true)
+            .build();
+
+        public static final ConfigurationProperty<String> PROXY_HOST = ConfigurationUtils.stringSharedPropertyBuilder("http.proxy.hostname")
+            .canLogValue(true)
+            .build();
+
+        public static final ConfigurationProperty<Integer> PROXY_PORT = ConfigurationUtils.integerSharedPropertyBuilder("http.proxy.port")
+            .build();
+
+        public static final ConfigurationProperty<String> PROXY_USER = ConfigurationUtils.stringSharedPropertyBuilder("http.proxy.username")
+            .canLogValue(true)
+            .build();
+
+        public static final ConfigurationProperty<String> PROXY_PASSWORD = ConfigurationUtils.stringSharedPropertyBuilder("http.proxy.password").build();
+
+        private static final int DEFAULT_HTTPS_PORT = 443;
+        private static final int DEFAULT_HTTP_PORT = 80;
+
+
+        public static String getProxyHost(Configuration configuration, boolean isHttps) {
+            String host = configuration.get(PROXY_HOST);
+            if (host == null) {
+                host = configuration.get(isHttps ? "https.proxyHost" : "http.proxyHost");
+            }
```

Comment:
This concept is interesting to me, so it's possible to have cross-configuration here where some of the properties we inspect as set and used but others could fallback to the Java system properties, ex:

`http.proxy.non-proxy-hosts = "non proxy host regex"`
`http.proxyUser` = "proxy-user"`

Is this something we'd want? Or do we want to be stricter in you're either using

- Environment variable proxy
- Java system properties proxy
- Azure SDKs for Java system properties proxy

Where you must configure everything based on one of those sets of environment variables/system properties.

Reply:
ok, so I've separated explicit source, environment source and system property source:
- the existing `Configuration.get(String)` retrieves config from sys properties or env vars. It **does not** support reading new config from file - it's only supported with `Configuration.get(ConfigurationProperty)`
- `ConfigurationProperty` has different setters for sys props and env vars, which can only be read from corresponding source and not the other.